x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorScheduling.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.application.connector;
+
+import org.elasticsearch.ElasticsearchParseException;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.xcontent.XContentHelper;
+import org.elasticsearch.xcontent.ConstructingObjectParser;
+import org.elasticsearch.xcontent.ObjectParser;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentParser;
+import org.elasticsearch.xcontent.XContentParserConfiguration;
+import org.elasticsearch.xcontent.XContentType;
+
+import java.io.IOException;
+
+import static org.elasticsearch.xcontent.ConstructingObjectParser.constructorArg;
+
+public class ConnectorScheduling implements Writeable, ToXContentObject {
+
+    private final ScheduleConfig accessControl;
+    private final ScheduleConfig full;
+    private final ScheduleConfig incremental;
+
+    public static final ParseField ACCESS_CONTROL_FIELD = new ParseField("access_control");
+    public static final ParseField FULL_FIELD = new ParseField("full");
+    public static final ParseField INCREMENTAL_FIELD = new ParseField("incremental");
+
+    // Constructors, getters, and setters for ConnectorScheduling
+    private ConnectorScheduling(ScheduleConfig accessControl, ScheduleConfig full, ScheduleConfig incremental) {
+        this.accessControl = accessControl;
+        this.full = full;
+        this.incremental = incremental;
+    }
+
+    public ConnectorScheduling(StreamInput in) throws IOException {
+        this.accessControl = readScheduleConfigFromStream(in);
```

Comment:
You can use the StreamInput constructors for the classes to read directly from the StreamInput:

```suggestion
        this.accessControl = new ScheduleConfig(in);
```

as the constructors consume from the StreamInput, it's equivalent to the methods you've defined here 👍